### PR TITLE
[ADD] Adding module account_invoice_number - First commit

### PR DESCRIPTION
**Pull request by haidd**
Expected behavior:

These information should appear on:

- Customer Invoice
- Supplier Invoice
- Sale Receipt (type is Pay Later)
- Purchase Receipt (type is Pay Later)

Solution:
Create 3 fields Invoice Number, Form, Serie on:

- Customer Invoice
- Supplier Invoice
- Sale Receipt (type is Pay Later)
- Purchase Receipt (type is Pay Later)

Invoice Number: integer
Form: char, 10 characters
Serie: char, 10 characters
It is required to have these fields input when:

- Validating customer/supplier invoice

### CODE REVIEW

File: README.md
Comment (jcdrubay) [inline README.md]: rename to l10n_vn_account_invoice

File: account_invoice_number/__manifest__.py
Comment (jcdrubay) [inline account_invoice_number/__manifest__.py]: Version should be 10.0.1.0.0

File: account_invoice_number/__manifest__.py
Comment (jcdrubay) [inline account_invoice_number/__manifest__.py]: Ensure that your follow templates: https://github.com/OCA/maintainer-tools/tree/master/template/module

There should not be description.

File: account_invoice_number/__manifest__.py
Comment (jcdrubay) [inline account_invoice_number/__manifest__.py]: Authors: need to add OCA     "author": "<AUTHOR(S)>, Odoo Community Association (OCA)",

File: account_invoice_number/models/account_invoice.py
Comment (jcdrubay) [inline account_invoice_number/models/account_invoice.py]: Remove the string  label if not necessary

Rename to invoice_number_official ?

File: account_invoice_number/models/account_voucher.py
Comment (jcdrubay) [inline account_invoice_number/models/account_voucher.py]: Maybe we should rename to voucher_number_official ?

File: account_invoice_number/views/account_invoice_view.xml
Comment (jcdrubay) [inline account_invoice_number/views/account_invoice_view.xml]: Remove `<data>`

**Comment by sbidoul** [comment]
I suggest you name this module l10n_vn_account_invoice_number, or if it has a generic character, propose it to oca/account-invoicing.

**Comment by haidd** [comment]
Thanks for your suggestion, @sbidoul. I will check it

File: l10n_vn_account_invoice_number/__init__.py
Comment (jcdrubay) [inline l10n_vn_account_invoice_number/__init__.py]: from . import models

File: l10n_vn_account_invoice_number/models/__init__.py
Comment (jcdrubay) [inline l10n_vn_account_invoice_number/models/__init__.py]: from . import account_invoice
from . import account_voucher

**Review by jcdrubay — CHANGES_REQUESTED** [review]

File: l10n_vn_account_invoice_number/README.rst
Comment (jcdrubay) [inline l10n_vn_account_invoice_number/README.rst:15]: > (type is Pay Later)

What does it mean?

File: l10n_vn_account_invoice_number/README.rst
Comment (jcdrubay) [inline l10n_vn_account_invoice_number/README.rst:22]: Customer > Customer Invoice

File: l10n_vn_account_invoice_number/README.rst
Comment (jcdrubay) [inline l10n_vn_account_invoice_number/README.rst:36]: Add info... I think something like:

```
To use this module, you need to create / validate any of these:
 - Customer Invoice
 - Supplier Invoice
 - Sales Receipt
 - Purchase Receipt
When validating, a control will ensure that the fields added by this module are being set.
```

File: l10n_vn_account_invoice_number/README.rst
Comment (jcdrubay) [inline l10n_vn_account_invoice_number/README.rst:42]: This should be removed

File: l10n_vn_account_invoice_number/__init__.py
Comment (jcdrubay) [inline l10n_vn_account_invoice_number/__init__.py:2]: `# Copyright 2017 Trobz (http://trobz.com)`

Copyright cannot start  before you write the code. 
Please use the format from the templates: https://github.com/OCA/maintainer-tools/blob/master/template/module/models/model_name.py

Please fix for all files

File: l10n_vn_account_invoice_number/__manifest__.py
Comment (jcdrubay) [inline l10n_vn_account_invoice_number/__manifest__.py:20]: remove the suffix `_view.xml` 

https://github.com/OCA/maintainer-tools/blob/master/CONTRIBUTING.md#file-naming

File: l10n_vn_account_invoice_number/views/account_invoice_view.xml
Comment (jcdrubay) [inline l10n_vn_account_invoice_number/views/account_invoice_view.xml:4]: Should be `id="invoice_supplier_form"`

Please check other xml_ids based on:
 https://github.com/OCA/maintainer-tools/blob/master/CONTRIBUTING.md#security-view-and-action
https://github.com/OCA/maintainer-tools/blob/master/CONTRIBUTING.md#security-view-and-action

File: l10n_vn_account_invoice_number/README.rst
Comment (jcdrubay) [inline l10n_vn_account_invoice_number/README.rst:16]: If there is an invoice number why not using an invoice?

File: l10n_vn_account_invoice_number/__manifest__.py
Comment (jcdrubay) [inline l10n_vn_account_invoice_number/__manifest__.py:17]: We should remove dependencies to sale, purchase.

I believe that account_voucher can also be removed if we agree that it is not a receipt when there is an invoice number.

File: l10n_vn_account_invoice_number/models/account_invoice.py
Comment (jcdrubay) [inline l10n_vn_account_invoice_number/models/account_invoice.py:14]: Remove size.
For sure it is not supposed to be 10 characters.

File: l10n_vn_account_invoice_number/models/account_invoice.py
Comment (jcdrubay) [inline l10n_vn_account_invoice_number/models/account_invoice.py:13]: I think we can remove the size, why do you specify? In case the number changes to 11 characters we will need to change the code...

I think it is wrong with 10, I have an invoice form with 11 characters: 01GTKT2/009

File: l10n_vn_account_invoice_number/models/account_invoice.py
Comment (jcdrubay) [inline l10n_vn_account_invoice_number/models/account_invoice.py:12]: As help message we should put the Vietnamese name of those 3 fields (and we should also mention the Vietnamese names in the README of the module).

Maybe we should rename the field invoice_number into `tax_invoice_number`, then `tax_invoice_form` and `tax_invoice_serie`

**Review by jcdrubay — CHANGES_REQUESTED** [review]

**Comment by fanha99** [comment]
hi, could you please tell me what is the module for?
01GTKT3/002 TH/16P 0000579?

**Comment by fanha99** [comment]
Are you going to continue?

**Comment by jcdrubay** [comment]
Hello @fanha99 , we will continue on this very soon if there is no feedback for original contributor.

File: l10n_vn_account_invoice_number/views/account_invoice_view.xml
Comment (jcdrubay) [inline l10n_vn_account_invoice_number/views/account_invoice_view.xml:10]: For all views, hide those 3 fields when the related company_id is not in Vietnam, this will be useful in the case of cross-country multicompany.

**Review by jcdrubay — CHANGES_REQUESTED** [review]

File: l10n_vn_account_invoice_number/views/account_voucher_view.xml
Comment (fanha99) [inline l10n_vn_account_invoice_number/views/account_voucher_view.xml:11]: why are you hide the tax invoice number when the choosing pay_now?

**Review by fanha99 — COMMENTED** [review]
question:

**Comment by jcdrubay** [comment]
Follow-up in #3

**Comment by haidd** [comment]
Hello @fanha99 , sorry for late reply!
At the moment, we are following up on [OCV - Localization for Vietnamese Accounting Road Map](https://goo.gl/bqfYU2), please put your comments, your ideas into that documentation and we can discuss together :+1:  Anw, thanks for your #3 :100: :+1:

**Comment by jcdrubay** [comment]
IMHO...

That documents is used to pick-up ideas, I don't think it is used for any kind of follow-up.

Ideally, when someone wants to start working on one point of that document he/she should create an issue describing what would be done giving a chance for some inputs before code starts.

Then would come the merge request.

Once the issue or merge request is created in github, then github should become the reference.

**Comment by jcdrubay** [comment]
Closing in favor of #3

**Pull request closed without merging**
